test/Microsoft.AspNet.Razor.Test/TestFiles/CodeGenerator/CS/Output/NullConditionalExpressions.DesignTime.cs

```diff
@@ -0,0 +1,99 @@
+namespace TestOutput
+{
+    using System;
+    using System.Threading.Tasks;
+
+    public class NullConditionalExpressions
+    {
+        private static object @__o;
+        private void @__RazorDesignTimeHelpers__()
+        {
+            #pragma warning disable 219
+            #pragma warning restore 219
+        }
+        #line hidden
+        public NullConditionalExpressions()
+        {
+        }
+
+        #pragma warning disable 1998
+        public override async Task ExecuteAsync()
+        {
+#line 1 "NullConditionalExpressions.cshtml"
+  
+    
+
+#line default
+#line hidden
+
+#line 2 "NullConditionalExpressions.cshtml"
+__o = ViewBag?.Data;
+
+#line default
+#line hidden
+#line 2 "NullConditionalExpressions.cshtml"
+                  
+    
+
+#line default
+#line hidden
+
+#line 3 "NullConditionalExpressions.cshtml"
+__o = ViewBag.IntIndexer?[0];
+
+#line default
+#line hidden
+#line 3 "NullConditionalExpressions.cshtml"
+                           
+    
+
+#line default
+#line hidden
+
+#line 4 "NullConditionalExpressions.cshtml"
+__o = ViewBag.StrIndexer?["key"];
+
+#line default
+#line hidden
+#line 4 "NullConditionalExpressions.cshtml"
+                               
+    
+
+#line default
+#line hidden
+
+#line 5 "NullConditionalExpressions.cshtml"
+__o = ViewBag?.Method(Value?[23]?.More)(SecondValue)?["key"];
```

Comment:
Not something that should change in this PR but might be worth filing a bug to remove this corner case from implicit expression support. Adjacent parenthesized expressions happen only when you have a `Func<..., Action<...>>`, `Func<..., Func<...>>` or similar nested delegates and you're calling the whole thing in one expression. Seems more likely the expression ends after the first method invocation and the next part begins a new parenthesized sentence.

Fix relative to this PR's version of the files would be at https://github.com/aspnet/Razor/pull/396/files#diff-d2cd6cc81ade624cba92cccb004b6afdR389